opsgenie/resource_opsgenie_service_incident_rule_test.go

```diff
@@ -0,0 +1,236 @@
+	}
+
+	for _, svc := range resp.Services {
+		if strings.HasPrefix(svc.Name, "genietest-") {
+			log.Printf("Destroying service %s", svc.Name)
+
+			deleteRequest := service.DeleteRequest{
+				Id: svc.Id,
+			}
+
+			if _, err := client.Delete(context.Background(), &deleteRequest); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func TestAccOpsGenieServiceIncidentRule_basic(t *testing.T) {
+	randomTeam := acctest.RandString(6)
+	randomService := acctest.RandString(6)
+
+	config := testAccOpsGenieServiceIncidentRule_basic(randomTeam, randomService)
+
+	resource.Test(t, resource.TestCase{
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckOpsGenieServiceIncidentRuleDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckOpsGenieServiceIncidentRuleExists("opsgenie_service_incident_rule.test"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccOpsGenieServiceIncidentRule_complete(t *testing.T) {
+	randomTeam := acctest.RandString(6)
+	randomService := acctest.RandString(6)
+	randomDescription := acctest.RandString(20)
+
+	config := testAccOpsGenieServiceIncidentRule_complete(randomTeam, randomService, randomDescription)
+
+	resource.Test(t, resource.TestCase{
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckOpsGenieServiceIncidentRuleDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckOpsGenieServiceIncidentRuleExists("opsgenie_service_incident_rule.test2"),
+				),
+			},
+		},
+	})
+}
+
+func testCheckOpsGenieServiceIncidentRuleDestroy(s *terraform.State) error {
+	client, err := service.NewClient(testAccProvider.Meta().(*OpsgenieClient).client.Config)
+	if err != nil {
+		return err
+	}
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "opsgenie_service" {
+			continue
+		}
+		req := service.GetRequest{
+			Id: rs.Primary.Attributes["id"],
+		}
+		_, err := client.Get(context.Background(), &req)
+		if err != nil {
+			x := err.(*ogClient.ApiError)
+			if x.StatusCode != 404 {
+				return errors.New(fmt.Sprintf("Service still exists : %s", x.Error()))
+			}
+		}
+	}
+
+	return nil
+}
+
+func testCheckOpsGenieServiceIncidentRuleExists(name string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
+		}
+
+		service_incident_rule_id := rs.Primary.Attributes["id"]
+		service_id := rs.Primary.Attributes["service_id"]
+
+		client, err := service.NewClient(testAccProvider.Meta().(*OpsgenieClient).client.Config)
+		if err != nil {
+			return err
+		}
+		incident_rule_res, err := client.GetIncidentRules(context.Background(), &service.GetIncidentRulesRequest{
+			ServiceId: service_id,
+		})
+
+		if err != nil {
+			return fmt.Errorf("Bad: Service ID %q does not exist", service_id)
+		} else {
+			fmt.Println("SUNNY Error is NIL, moving forward")
```

Comment:
also this line should be generic of course

Reply:
Changed